ocean-climate-simulation/ocean_climate_simulation_mlir.jl

```
@@ -19,15 +19,47 @@ function loop!(model, Ninner)
     return nothing
 end
 
+# Pre-raise IR
+@info "Compiling before raise kernel..."
+before_raise = try
+    @code_hlo optimize=:before_raise raise=true loop!(model, 3)
+catch e
+    @error "Failed to compile" exception=(e, catch_backtrace())
+    Text("""
+    // Failed to compile
+    //$e
+    """)
```

Should we throw an error here?  Otherwise we don't easily notice something failed without opening the log.

---

Yes, but if there is an error we cannot get any IR because the GA step fails. Is there a way to fail the action while still uploading files?

---

```
failed = false

try
    error()
catch err
    @error "Failed to compile" exception=(e, catch_backtrace())
    global failed = true
end

failed && error("oh no")
```